### ISSUE TITLE
Delay in Moving Task Cards in Board View

### ISSUE THREAD

**Issue by JustFxDev** [aljawaid/PluginManager#28]
Hi aljawaid,

thanks for your great contributions to Kanboard!

I tried the PluginManager. Looks great and furthermore provides important informations about plugins.

Unfortunately i have delayed movement of the task cards on the board when moving them around by pressing left mouse and moving the mouse beetween the columns. I doublechecked this behavior. It's gone as soon as i delete the plugin folder for PluginManager and clear the cache. If i copy the folder back, do a cache clean reload: it's back again.

I suppose a correlation to the css issue:

https://github.com/aljawaid/PluginManager/issues/27

Cheers, fx

**Comment by aljawaid** [comment]
@JustFxDev 

Hello, thanks for the message and I'm glad others are finding my my plugins useful.

About the issue, I can't reproduce it as in the CSS file there is nothing linked to the cards or tasks.  I will have a deeper look when I get home on the pc.

**Comment by creecros** [comment]
I would guess it's you js

**Comment by aljawaid** [comment]
@creecros thanks, I have made a change, I think if anything was the issue, it would be that.

@JustFxDev can you replace the `js` file with the one in this repository to test? or shall I just release a new version in the hope that the issue is fixed?

**Comment by JustFxDev** [comment]
Hi aljawaid,

i've download the latest zip (containing files with date 20.12.2022 - 21:13). I' am sorry, i can't see any difference. I delete the old folder, copied the new files completly, cleared all caches and tried.

I try to describe it further:

I pick a card by left mouse and drag it around. I have a heavy delay in the animation that shows up the gray shadow and the card which is behind that shadow moving around.

If i can help further just let me know!

Cheers, fx

**Comment by aljawaid** [comment]
I will keep an eye on it but for the time being I cannot reproduce the problem.

**Comment by JustFxDev** [comment]
If you have one or more hints which files / code sections / css section could contain the possible issue or the most common source: just let me know. I can try to investigate that further.

**Comment by aljawaid** [comment]
@JustFxDev I have changed the js code in the plugin, thats the only place where @creecros rightfully told me to look.  If you want, delete the JS hook section in `plugin.php` and try again.
Try deleting this:
```
$this->hook->on('template:layout:js', array('template' => 'plugins/PluginManager/Assets/js/plugin-manager-top-btn.js'));
```

I look at the rest of the code and it has nothing to do with tasks or boards.  To be honest, I dont think this plugin is the real cause of your issue

**Comment by creecros** [comment]
first step to reproduce an issue is to list your kb version, and all plugins you have installed. server config helps as well.

**Comment by creecros** [comment]
That said, I just tested, and OP is correct @aljawaid  there is an issue. I 100% confirm, it is coming from this plugin. I'll take a look when i get a chance.

**Comment by creecros** [comment]
I can also confirm it is 100% coming from your CSS, not the JS.

**Comment by creecros** [comment]
and after some tinkering, after I removed everything relataed to svg, it works fine again. I dont do css, but i think drawing svgs is tedious, seems where your issue is.

**Comment by aljawaid** [comment]
@creecros bugger, okay thanks... I will look into this

**Comment by aljawaid** [comment]
> and after some tinkering, after I removed everything relataed to svg, it works fine again. I dont do css, but i think drawing svgs is tedious, seems where your issue is.

- tinkering with the svg-related code in the css? or in one of the template files?
- the thing is, all my plugins use svg and i use kanboard daily (the plugins are for personal use, so they are used) and there doesnt seem to be any issues... i'm not saying that svg is not the problem

**Comment by aljawaid** [comment]
@JustFxDev could you please try again using the latest updates in the repository? (NOT the latest release)

I still dont see any/much of a delay on my side, but I'm hoping/suspecting that 9b1e2d5 was the issue. (If not, the next thing for me to look into will be the svg/css hacks in the css file, but again, they have nothing to do with tasks/boards.)

**Comment by creecros** [comment]
again, I know nothing about css, as for tinkering I deleted every block thay had the word svg in it. then it worked fine.

**Comment by aljawaid** [comment]
> again, I know nothing about css, as for tinkering I deleted every block that had the word svg in it. then it worked fine.

great, I can start on this.... Your hint is the word svg...

**Comment by aljawaid** [comment]
- [ ] Check for CSS Syntax Errors
- [ ] Check/Replace for CSS Icon/SVG Hacks (possibly, global dropdown menu css hack for plugins icon could be issue as thats the only bit of active code that could be 'ready to work' on the boards/tasks)
- [ ] Check/Replace all SVG CSS Code (comment out and test all svg related code... use as a complete last resort)

**Comment by JustFxDev** [comment]
@creecros is right: his gif(?) shows exactly the behavior that i have as soon as i copy the plugin to the plugins folder. 

I think we are aligned now: something in the plugin matters.

@aljawaid i saw your infos regarding svg (that checklist) should i wait until it is finished bevor testing on my own?

Thanks to both of you for your efforts. Great!

Cheers, fx

**Comment by aljawaid** [comment]
> @aljawaid i saw your infos regarding svg (that checklist) should i wait until it is finished bevor testing on my own?
> 

Yes please check now.  After your next comment, I will do item 2 on the list.  I am hopeful that item 1 will have solved it, if not it should be item 2.  The reason why I dont want to touch item 3 is because my other plugins use svg with no other issues and also, it will be a major re-write of code.

**Comment by aljawaid** [comment]
> Maybe this is a silly question: I download the latest Zip from the main page and i geht the latest changes, right?
> 
YES thats right

**Comment by aljawaid** [comment]
@JustFxDev try again after using the latest code, or apply this change and then try: bdcf4a0
- Make sure your cache is cleared

**Comment by JustFxDev** [comment]
I applied latest code (downloading the zip), copied it, cleared the cache and the issue stays still the same

**Comment by aljawaid** [comment]
> I applied latest code (downloading the zip), copied it, cleared the cache and the issue stays still the same

ok cool I will look into it further once I have got somewhere with KBColours plugin

**Comment by JustFxDev** [comment]
Hi aljawaid,

when i remove line 45 in `plugin-mananger.css` the moving around seams to work again. I'am not sure what implications this has to the plugin-manager plugin
```
div, a, details, summary, i, img, table, tr, .btn-uninstall, svg, #PluginSideBar > ul li, figure {
    transition: ease-in-out all .3s;
}

```

**Comment by aljawaid** [comment]
@JustFxDev Perfect, please do two things to test:

1. Test the code as:
```
div, a, details, summary, i, img, table, tr, .btn-uninstall, #PluginSideBar > ul li, figure {
    transition: ease-in-out all .3s;
}
```
2.  Test the code as:
```
div, a, details, summary, i, img, table, tr, .btn-uninstall, svg, figure {
    transition: ease-in-out all .3s;
}
```

Please let me know which one solves the problem...

**Comment by JustFxDev** [comment]
Both variantes do not fix that. I think the css is not precise enough in selecting the right target html. It is formulated too generally.

If you can tell me with html snippet(s) in plugin-manager should be affected by that i can try to help to remove that from that task board.

From my point of view ist should not be in this list:

![grafik](https://user-images.githubusercontent.com/82320201/209547734-4da173d9-0e71-4401-aefc-300982a657f9.png)

**Comment by aljawaid** [comment]
> From my point of view ist should not be in this list:
> 
Thats just the filename.  Anyhow, I commented out the block of code altogether.  You earlier said removing that line fixed your probem.  Please check with this latest change that I did.

**Comment by JustFxDev** [comment]
Well, thats not just the filename. What you see is the css applied to the board view of Kanboard. And if your plugin-manager.css is listed than this means one ore more css statements in that file apply to html ids or classes of that current page (and not only to plugin-manager views / pages).

Which means also: the css contained in that file is not "selective enough" to avoid to match other classes as well. Result: the browser assigns css to classes which where not meant to have them...

**Comment by aljawaid** [comment]
Yes you are correct, I was referring to the circle in your screenshot

**Comment by aljawaid** [comment]
I removed that block of code entirely for now. Please check f5ba607

**Comment by JustFxDev** [comment]
I installed the latest Zip from the main page. Cleared caches, installed and it works as expected :+1: :smiley:

**Comment by aljawaid** [comment]
Excellent. I will adjust that code and release a new version soon

**Comment by aljawaid** [comment]
@JustFxDev kindly try with the latest code.... I believe the problem was the `div` - that was too global

**Comment by JustFxDev** [comment]
Downloaded latest code (10 min. ago). Fine so far.

Thanks!

**Comment by aljawaid** [comment]
Fixed by f8c8e89
@JustFxDev thanks for testing
